### DISCUSSION TITLE
v4.37.2

### DISCUSSION THREAD

**Discussion by xiaokangwang** [v2fly/v2ray-core#880] [General]
## Features

* **Websocket**: support browser forwarder (#818)
  * Docs: [Websocket](https://www.v2fly.org/config/transport/websocket.html#websocketobject) & [BrowserForwarder](https://www.v2fly.org/config/browserforwarder.html)
  * Thanks @rprx
* **Websocket**: support Websocket 0-RTT early data (#818)
  * Docs: [Websocket](https://www.v2fly.org/config/transport/websocket.html#websocketobject)
  * Thanks @rprx
* **DNS**: add queryStrategy option for DNS (#794)
  * Docs: [DNS](https://www.v2fly.org/config/dns.html#dnsobject)
  * Thanks @Loyalsoldier
* **Shadowsocks**: add replay protection for Shadowsocks proxy (#777)
  * Docs: [Shadowsocks](https://www.v2fly.org/config/protocols/shadowsocks.html#outboundconfigurationobject)
  * Thanks @oif
* **Rule**: added inversed geoip matching. (https://github.com/v2fly/v2ray-core/pull/860) @Loyalsoldier
* **DNS**: add disableFallback & skipFallback option for DNS client. (#864) @Loyalsoldier

## Fixes

* **gRPC**: add grpcSettings & gunSettings to streamSettings (https://github.com/v2fly/v2ray-core/commit/00879c483e54dd327b153920569353c96398f90f) @rprx
* **Protobuf**: vprotogen loop dependency (#797) @Loyalsoldier @U-v-U
* **DNS**: tests timeout due to network instability (#805) @Loyalsoldier
* **DNS**: remove AA header flag in DNS query (#817) @Loyalsoldier
* **TProxy**: cannot find IPv6 destination in redirect mode (#815) @mzz2017
* **Context**: ctx initialization for core.functions (#841) @rurirei
* **Websocket**: Websocket early data 404 bug (#859) @k79e
* **Browser Forwarder**: cannot load resources on Windows (#849) @dyhkwong @kslr
* **QUIC**: disconnect due to timeout (#850) @bhoppi
* **FakeDNS** fixed fake DNS option. (https://github.com/v2fly/v2ray-core/pull/879) @sixg0000d @Loyalsoldier 

## Chores

* **Log**: remove package path prefix in logs (#840 https://github.com/v2fly/v2ray-core/commit/01380177b8f24b66a4f72031a7cef38da4753028 https://github.com/v2fly/v2ray-core/commit/78c19934313a206c29a41cb104524163fb535e5b) @kslr
* Update protobuf & dependencies & workflows & lint @Loyalsoldier
* Use Go v1.16.3

## Notices

* **VMess**: From Jan 1, 2022, compatibility for legacy VMess MD5 will be disabled by default. Visit [here](https://www.v2fly.org/config/protocols/vmess.html#vmess-md5-%E8%AE%A4%E8%AF%81%E4%BF%A1%E6%81%AF-%E6%B7%98%E6%B1%B0%E6%9C%BA%E5%88%B6) for more information.
* You are able to compile exactly the same binaries as the ones in Assets section below by simply following the [compiling guide](https://www.v2fly.org/developer/intro/compile.html).

## For downstream developers

The Go module name of `v2ray-core` has been changed to `github.com/v2fly/v2ray-core/v4`. Do NOT use `v2ray.com/core` anymore.

<hr /><em>This discussion was created from the release <a href='https://github.com/v2fly/v2ray-core/releases/tag/v4.37.2'>v4.37.2</a>.</em>

**Comment by rurirei** [comment]
what discussion 😕

**Reply by xiaokangwang** [reply]
This is a function in the Github I wants to try... So don't worry about that.

**Reply by xiaokangwang** [reply]
@rurirei Please have a look at https://github.com/v2fly/v2ray-core/issues/872 . I will need server side log to know what exactly goes wrong.

**Reply by rurirei** [reply]
> This is a function in the Github I wants to try... So don't worry about that.

🤣 ok.

**Comment by jimsmt** [comment]
相比于4.37.1修复了些什么呢

**Reply by k79e** [reply]
我也想知道
....看起来好像是fakedns的

**Reply by rurirei** [reply]
#879 #874 #864

**Reply by k79e** [reply]
这真是太实用了

**Comment by coconut-piglet** [comment]
服务端更新了之后发现以前是从`/usr/local/share/v2ray`读dat的现在变成了从可执行文件所在的地方`/usr/local/bin`读，脚本还是把dat放到了前一个目录所以更新完炸了（问题不大，链接过去就好了），是bug咩还是feature呀(:O)

**Reply by kslr** [reply]
什么系统，怎么安装的

**Reply by coconut-piglet** [reply]
Ubuntu 20.04，用fhs-install-v2ray脚本更新的

**Reply by kslr** [reply]
> Ubuntu 20.04，用fhs-install-v2ray脚本更新的

因为新的 release 文件中不再携带geosite, geoip 数据，需要时从网络下载，你有兴趣来解决这个问题吗

**Reply by coconut-piglet** [reply]
emmm不是太确定您的意思，我检查过releases页下载的zip中依旧含有geosite和geoip呀😲

是说从4.37.2版本起将从可执行文件所在目录读取geoip和geosite而非`/usr/local/share/v2ray`，即并不是bug，fhs-install-v2ray需要更新（或者用户自行设置`DAT_PATH`再跑脚本）；

还是说我遇到的情况并非预期行为，有bug要修复嘞？

**Reply by kslr** [reply]
是我记错了，和这个无关

**Reply by Loyalsoldier** [reply]
是个 bug，已修复 https://github.com/v2fly/v2ray-core/pull/889

**Comment by xhqpp** [comment]
更新到v4.37.2发现服务端无法启动，报错无法读取geoip，配置文件用的是v2ray-examples/VMess-TCP的模板，手动将geosite, geoip从/usr/local/share/v2ray复制到/usr/local/bin问题解决。

**Reply by Loyalsoldier** [reply]
已修复 https://github.com/v2fly/v2ray-core/pull/889
